foursight_core/checks/codebuild_checks.py

```diff
@@ -0,0 +1,73 @@
+from dcicutils.codebuild_utils import CodeBuildUtils
+from .helpers.confchecks import *
+
+
+@check_function()
+def codebuild_status(connection, **kwargs):
+    """ Returns list of available CodeBuild projects """
+    check = CheckResult(connection, 'codebuild_status')
+    client = CodeBuildUtils()
+    projects = client.list_projects()
+    check.status = 'PASS'
+    check.summary = 'See full output for info on valid build names'
+    check.full_output = projects
+    return check
+
+
+@check_function(build_name=None)
+def trigger_codebuild_run(connection, **kwargs):
+    """ This checks triggers a run of the CodeBuild pipeline
+        Usually, there is only 1 CodeBuild pipeline per account - this function
+        will work out of the box where there is only 1, and will require kwarg
+        if there are multiple.
+    """
+    check = CheckResult(connection, 'trigger_codebuild_run')
+    client = CodeBuildUtils()
+    projects = client.list_projects()
+    build_name = kwargs.get('build_name', None)
+    if len(projects) == 1:
+        resp = client.run_project_build(project_name=projects[0])
+        check.full_output = resp
+        check.status = 'PASS'
+        check.summary = f'Triggered build {projects[0]}'
+    elif build_name:
+        resp = client.run_project_build(project_name=build_name)
+        check.full_output = resp
+        check.status = 'PASS'
+        check.summary = f'Triggered build {build_name}'
```

Comment:
I'd like to see the two clauses consolidated, as in (untested):
```suggestion
    build_name = projects[0] if len(projects) == 1 else build_name
    if build_name:
        resp = client.run_project_build(project_name=build_name)
        check.full_output = resp
        check.status = 'PASS'
        check.summary = f'Triggered build {build_name}'
```
although why is `build_name` not preferred if supplied? I sort of imagine I'd rather see (again untested):
```
build_name = build_name or projects[0]
```